widgets/tokens/src/TokensTableHeader/CreateTokenModal.tsx

```diff
@@ -1,37 +1,64 @@
+import { isEqual } from 'lodash';
 import { useState } from 'react';
 import { flushSync } from 'react-dom';
 import { translationPath } from '../widget.consts';
 import { tableRefreshEvent } from '../TokensTable.consts';
 import { RequestStatus } from '../types';
 import CreatedToken from './CreatedToken';
-import type { ReceivedToken } from './CreateTokenModal.types';
+import type { ReceivedToken, TokensPostRequestBody } from './CreateTokenModal.types';
 
-const { useInput } = Stage.Hooks;
+const { useInput, useErrors } = Stage.Hooks;
 const { getT } = Stage.Utils;
-const { Modal, Icon, CancelButton, ApproveButton, Input, Form, LoadingOverlay, Message } = Stage.Basic;
+const { Modal, Icon, CancelButton, ApproveButton, DateInput, Input, Form, LoadingOverlay, Message } = Stage.Basic;
 
 const t = getT(`${translationPath}.createModal`);
+const expirationDateFormat = 'YYYY-MM-DD HH:mm';
+
+function getTokensPostRequestBody(description: string, expirationDate: string): TokensPostRequestBody {
+    const body: TokensPostRequestBody = {};
+    if (description) body.description = description;
+    if (expirationDate) body.expiration_date = moment.utc(moment(expirationDate)).format(expirationDateFormat);
+    return body;
+}
 
 interface CreateTokenModalProps {
     onClose: () => void;
     toolbox: Stage.Types.Toolbox;
 }
 
 const CreateTokenModal = ({ onClose, toolbox }: CreateTokenModalProps) => {
+    const { setErrors, getContextError, clearErrors } = useErrors();
     const [description, setDescription] = useInput('');
+    const [expirationDate, setExpirationDate] = useInput('');
     const [submittingStatus, setSubmittingStatus] = useState<RequestStatus>(RequestStatus.INITIAL);
     const [receivedToken, setReceivedToken] = useState<ReceivedToken>();
     const showCreateForm = submittingStatus !== RequestStatus.SUBMITTED;
     const manager = toolbox.getManager();
 
     const handleSubmit = () => {
+        clearErrors();
+
+        if (expirationDate) {
+            const expirationDateMoment = moment(expirationDate);
+            const expirationDateHasInvalidFormat =
+                !expirationDateMoment.isValid() ||
+                !isEqual(expirationDateMoment.format(expirationDateFormat), expirationDate);
+
+            if (expirationDateHasInvalidFormat) {
+                setErrors({ expirationDate: t('errors.expirationDateInvalidFormat') });
+                return;
+            }
+
+            if (expirationDateMoment.isBefore(moment())) {
```

Comment:
We can extract additional condition as a variable, to make the if'ology more readable 🎉 

```suggestion
            const expirationDateIsPast = expirationDateMoment.isBefore(moment());
            
            if (expirationDateHasInvalidFormat) {
                setErrors({ expirationDate: t('errors.expirationDateInvalidFormat') });
                return;
            }

            if (expirationDateIsPast) {
```

Reply:
Didn't do that intentionally as I thought that `expirationDateMoment.isBefore(moment())` is clear enough. But OK, I'll change it.